include/nuttx/atomic.h

```diff
@@ -60,8 +60,14 @@ extern "C++"
 #    include <stdatomic.h>
 #    define ATOMIC_FUNC(f, n) atomic_##f##_explicit
 
-  typedef volatile _Atomic int32_t atomic_t;
-  typedef volatile _Atomic int64_t atomic64_t;
+#    if defined(__cplusplus)
```

Comment:
but why C++ code does not go through the branch from line 33

Reply:
Exactly because when compiling without the C++ standard includes (-nostdinc++), there is no "atomic" include available (this is the c++ standard include). "#if  __has_include(<atomic>)" will fail already.

But that's fine, the atomic functions from "stdatomic.h" work just fine also in C++; the only issues are that there is no __auto_type and parameters in C++ may not be "volatile". So it can use the compiler built-in atomic functions, just the parameter types need to be C++ compatible.